independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Identified.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.arc;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import javax.enterprise.util.AnnotationLiteral;
+import javax.inject.Qualifier;
+
+/**
+ * Qualify a bean with a string-based identifier.
```

Comment:
Maybe drop this and use https://github.com/smallrye/smallrye-common/blob/main/annotation/src/main/java/io/smallrye/common/annotation/Identifier.java?

Reply:
Yeah, it's the same semantics but we don't want to add a dependency on `smallrye-common`.

Reply:
I'm just thinking we probably shouldn't have 2 annotations with the same meaning available as part of public API -- and IIUC, this `@Identified` is meant to be public API?

Reply:
> this @Identified is meant to be public API

Yes, the `io.quarkus.arc` package is meant to be public API. Ok, let me try to add the `smallrye-common-annotation` dependency.

Reply:
Actually, we shouldn't use a public API annotation at all. It should be an internal qualifier instead. Because otherwise we could run into the same problem - a collision with user-provided qualifier.

Reply:
I've moved the qualifier annotation into the `io.quarkus.arc.impl` package.

Reply:
Ah good point.